spring-core/src/main/java/org/togglz/spring/activation/SpringEnvironmentPropertyActivationStrategy.java

```diff
@@ -0,0 +1,71 @@
+package org.togglz.spring.activation;
+
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.env.Environment;
+import org.togglz.core.Feature;
+import org.togglz.core.activation.Parameter;
+import org.togglz.core.activation.ParameterBuilder;
+import org.togglz.core.repository.FeatureState;
+import org.togglz.core.spi.ActivationStrategy;
+import org.togglz.core.user.FeatureUser;
+import org.togglz.core.util.Strings;
+import org.togglz.spring.util.ContextClassLoaderApplicationContextHolder;
+
+/**
+ * <p>
+ * An activation strategy based on the values of properties accessible within the Spring environment.
+ * </p>
+ * <p>
+ * It can either be based on a given property name, passed as a parameter, or a property name constructed from the
+ * {@link Feature} itself (e.g. {@code com.example.MyFeatures.FEATURE_NAME}).
+ * </p>
+ *
+ * @author Alasdair Mercer
+ */
+public class SpringEnvironmentPropertyActivationStrategy implements ActivationStrategy {
+
+    public static final String ID = "spring-environment-property";
+    static final String NAME = "Spring Environment Property";
+    public static final String PARAM_NAME = "name";
+
+    private static String getPropertyName(FeatureState featureState) {
+        String propertyName = featureState.getParameter(PARAM_NAME);
+        if (Strings.isNotBlank(propertyName)) {
+            return propertyName;
+        }
+
+        Feature feature = featureState.getFeature();
+        return feature.getClass().getName() + "." + feature.name();
+    }
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public boolean isActive(FeatureState featureState, FeatureUser user) {
+        ApplicationContext applicationContext = ContextClassLoaderApplicationContextHolder.get();
+        if (applicationContext == null) {
+            return false;
```

Comment:
Hmmm. Interesting question. Back then I handled the missing `HttpServletRequest` very gracefully because I didn't want to throw an exception if the user for some reason uses the toggle outside of a web request (like in a Quartz job for example).

However, we could consider a missing ApplicationContext in a Spring application as a major problem because in this case there is typically something wrong with your setup. So throwing an exception would be OK here.